airflow/contrib/hooks/databricks_hook.py

```diff
@@ -0,0 +1,190 @@
+        if method == 'GET':
+            request_func = requests.get
+        elif method == 'POST':
+            request_func = requests.post
+        else:
+            raise AirflowException('Unexpected HTTP Method: ' + method)
+
+        for attempt_num in range(1, self.retry_limit+1):
+            try:
+                response = request_func(
+                    url,
+                    json=api_parameters,
+                    auth=auth,
+                    headers=USER_AGENT_HEADER,
+                    timeout=self.timeout_seconds,
+                    verify=False)
+                if response.ok:
+                    return response.json()
+                else:
+                    # In this case, the user probably made a mistake.
+                    # Don't retry.
+                    raise AirflowException(response.content)
+            except (requests_exceptions.ConnectionError,
+                    requests_exceptions.Timeout) as e:
+                logging.error(('Attempt {0} API Request to Databricks failed ' +
+                              'with reason: {1}').format(attempt_num, e))
+        raise AirflowException(('API requests to Databricks failed {} times. ' +
+                               'Giving up.').format(self.retry_limit))
+
+    def submit_run(self, json):
+        """
+        Utility function to call the ``api/2.0/jobs/runs/submit`` endpoint.
+
+        :param json: The data used in the body of the request to the ``submit`` endpoint.
+        :type json: dict
+        """
+        response = self._do_api_call(SUBMIT_RUN_ENDPOINT, json)
+        return response['run_id']
+
+    def get_run_page_url(self, run_id):
+        api_params = {'run_id': run_id}
+        response = self._do_api_call(GET_RUN_ENDPOINT, api_params)
+        return response['run_page_url']
+
+    def get_run_state(self, run_id):
+        api_params = {'run_id': run_id}
+        response = self._do_api_call(GET_RUN_ENDPOINT, api_params)
+        life_cycle_state = response['state']['life_cycle_state']
+        # result_state may not be in the state if not terminal
+        result_state = response['state'].get('result_state', None)
+        state_message = response['state']['state_message']
+        return RunState(life_cycle_state, result_state, state_message)
+
+
+RUN_LIFE_CYCLE_STATES = [
+    'PENDING',
+    'RUNNING',
+    'TERMINATING',
+    'TERMINATED',
+    'SKIPPED',
+    'INTERNAL_ERROR'
+]
+
+
+class RunState:
+    """
+    Utility class for the run state concept of Databricks runs.
+    """
+    def __init__(self, life_cycle_state, result_state, state_message):
+        self.life_cycle_state = life_cycle_state
+        self.result_state = result_state
+        self.state_message = state_message
+
+    @property
+    def is_terminal(self):
+        if self.life_cycle_state not in RUN_LIFE_CYCLE_STATES:
+            raise AirflowException('Unexpected life cycle state: {}:'.format(
+                self.life_cycle_state))
```

Comment:
This is usually not a user error. So users do not know what to do if they hit this line. Hopefully we won't add new lifecycle states, or we will update this operator first. Then it is safe to say `if the state is introduced recently, please check Databricks user guide and upgrade Airflow if needed`.